### PR TITLE
Extract Protein module from MS2

### PR DESCRIPTION
**Pull request by labkey-adam**
#### Rationale
Some modules require protein services but don't need all of MS2. https://www.labkey.org/home/Developer/issues/Secure/issues-details.view?issueId=49958

### CODE REVIEW

File: protein/src/org/labkey/protein/ProteinContainerListener.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import org.jetbrains.annotations.NotNull;
+import org.labkey.api.data.Container;
+import org.labkey.api.data.ContainerManager.ContainerListener;
+import org.labkey.api.security.User;
+
+import java.beans.PropertyChangeEvent;
+import java.util.Collection;
+import java.util.Collections;
+
+public class ProteinContainerListener implements ContainerListener
```
Comment (labkey-jeckels) [inline protein/src/org/labkey/protein/ProteinContainerListener.java:29]: This looks like a bunch of no-ops. Is it needed?

File: protein/src/org/labkey/protein/ProteinModule.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    @NotNull
+    protected Collection<WebPartFactory> createWebPartFactories()
+    {
+        return Collections.emptyList();
+    }
+
+    @Override
+    protected void init()
+    {
+//        addController(ProteinController.NAME, ProteinController.class);
```
Comment (labkey-jeckels) [inline protein/src/org/labkey/protein/ProteinModule.java]: Is this a TODO too?

File: protein/api-src/org/labkey/api/protein/ProteomicsModule.java [hunk 3]
```diff
@@ -1,29 +1,25 @@
...
-    {
-        super(s);
-    }
-}
+import org.labkey.api.module.Module;
+
+/**
+ * Tagging interface for module that need to make sure that the protein table is available.
+ */
+public interface ProteomicsModule extends Module
```
Comment (labkey-jeckels) [inline protein/api-src/org/labkey/api/protein/ProteomicsModule.java:23]: Seems like a strange sequence in the refactor - moving a file and then repurposing it for a different class. Fine, just strange.

File: ms2/src/org/labkey/ms2/MS2Module.java [hunk 4]
```diff
@@ -264,41 +262,43 @@ protected void startupAfterSpringConfig(ModuleContext moduleContext)
...
+            fcs.addFileListener(new TableUpdaterFileListener(ProteinSchema.getTableInfoAnnotInsertions(), "FileName", TableUpdaterFileListener.Type.filePath, "InsertId"));
+            fcs.addFileListener(new TableUpdaterFileListener(ProteinSchema.getTableInfoFastaFiles(), "FileName", TableUpdaterFileListener.Type.filePath, "FastaId"));
+        }
 
-        UsageMetricsService.get().registerUsageMetrics(getName(), () -> {
-                Map<String, Object> results = new HashMap<>();
-                results.put("hasGeneOntologyData", new TableSelector(ProteinSchema.getTableInfoGoTerm()).exists());
-                return results;
-        });
+        UsageMetricsService.get().registerUsageMetrics(getName(), () -> Map.of("hasGeneOntologyData", new TableSelector(ProteinSchema.getTableInfoGoTerm()).exists()));
```
Comment (labkey-jeckels) [inline ms2/src/org/labkey/ms2/MS2Module.java]: This can probably switch over to the new ProteinModule

**Review by labkey-jeckels — APPROVED** [review]

File: protein/src/org/labkey/protein/ProteinContainerListener.java (hunk 1 above)
Comment (labkey-adam) [reply protein/src/org/labkey/protein/ProteinContainerListener.java:29]: Protein schema deletes will be moved here once this module manages the `prot` schema. Added a comment.

File: protein/src/org/labkey/protein/ProteinModule.java (hunk 2 above)
Comment (labkey-adam) [reply protein/src/org/labkey/protein/ProteinModule.java]: This has been moved.

File: ms2/src/org/labkey/ms2/MS2Module.java (hunk 4 above)
Comment (labkey-adam) [reply ms2/src/org/labkey/ms2/MS2Module.java]: This has been moved

File: protein/api-src/org/labkey/api/protein/ProteomicsModule.java (hunk 3 above)
Comment (labkey-adam) [reply protein/api-src/org/labkey/api/protein/ProteomicsModule.java:23]: Not my doing... Git is just confusing a deleted class with a moved class

**Pull request merged**